File: backend/policourt-backend/src/main/java/com/policourt/api/club/domain/model/Club.java
package com.policourt.api.club.domain.model;

import java.time.OffsetDateTime;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import com.policourt.api.shared.enums.GeneralStatus;
import com.policourt.api.sport.domain.model.Sport;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Club {
    private Long id;
    private String slug;
    private String name;
    private String description;
    private String imgUrl;
    private Sport sport;
    private GeneralStatus status;
    private Boolean isActive;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: frontend/src/features/types/club/Club.ts
export type ClubStatus = 'PUBLISHED' | 'DRAFT' | 'ARCHIVED' | 'SUSPENDED';

export interface Club {
    name: string;
    slug: string;
    description?: string;
    imgUrl?: string;
    status: ClubStatus;
    isActive: boolean;
    sportSlug: string;
}
